src/types/event-log.ts

```diff
@@ -23,16 +24,25 @@ export interface EventLog {
    * adds an event to a tenant's event log
    * @param tenant - the tenant's DID
    * @param messageCid - the CID of the message
+   * @param indexes - (key-value pairs) to be included as part of indexing this event.
    * @returns {Promise<string>} watermark
    */
-  append(tenant: string, messageCid: string): Promise<string>
+  append(tenant: string, messageCid: string, indexes: { [key:string]: string|boolean }): Promise<string>
 
   /**
    * retrieves all of a tenant's events that occurred after the watermark provided.
    * If no watermark is provided, all events for a given tenant will be returned.
    */
   getEvents(tenant: string, options?: GetEventsOptions): Promise<Array<Event>>
 
+  /**
+   * retrieves a filtered set of events that occurred after a the watermark provided.
+   * accepts multiple filters each with their own watermark.
+   *
+   * If no watermark is provided, all events for a given tenant and filter combo will be returned.
```

Comment:
Sanity: I thought cursor can only be `messageCid`? Is it not the case here and watermark should be used instead?

Reply:
The cursor can only be `messageCid` for MessageStore, it's the `watermark` value for EventLog.

In IndexLevel, the cursor can be whatever is chosen as the `itemId`.

I have to think about it a bit more, but thinking about it now that this has been generalized more... I'm not sure that we need to return the watermark at all, it might be internal and the messageCid itself is the cursor for this as well.

Reply:
This is great, I had a comment exactly on this later in the PR. Seems we are arriving at the same conclusion that `index.put()` signature might not need to be modified. Happy to jump on a call to confirm each other's understanding if needed.